File: test/corpus/component/04-no-color-ascii/case.go
package main

import (
	"fmt"

	"github.com/muesli/termenv"
)

func main() {
	// Force Ascii profile for consistent testing
	out := termenv.NewOutput(nil, termenv.WithProfile(termenv.Ascii))
	
	// Test NoColor with various color attempts
	fmt.Println("--- NoColor Tests (ASCII Profile) ---")
	
	// Test with RGB colors - should render as plain text
	red := out.String("Red Text").Foreground(termenv.RGBColor("#FF0000"))
	blue := out.String("Blue Text").Foreground(termenv.RGBColor("#0000FF"))
	green := out.String("Green Text").Foreground(termenv.RGBColor("#00FF00"))
	
	fmt.Printf("RGB Red: %s\n", red)
	fmt.Printf("RGB Blue: %s\n", blue)
	fmt.Printf("RGB Green: %s\n", green)
	
	// Test with ANSI colors - should render as plain text
	ansiRed := out.String("ANSI Red").Foreground(termenv.ANSIColor(1))
	ansiBlue := out.String("ANSI Blue").Foreground(termenv.ANSIColor(4))
	
	fmt.Printf("ANSI Red: %s\n", ansiRed)
	fmt.Printf("ANSI Blue: %s\n", ansiBlue)
	
	// Test with ANSI256 colors - should render as plain text
	ansi256Red := out.String("ANSI256 Red").Foreground(termenv.ANSI256Color(196))
	ansi256Blue := out.String("ANSI256 Blue").Foreground(termenv.ANSI256Color(21))
	
	fmt.Printf("ANSI256 Red: %s\n", ansi256Red)
	fmt.Printf("ANSI256 Blue: %s\n", ansi256Blue)
	
	// Test styling - should also render as plain text
	bold := out.String("Bold Text").Bold()
	italic := out.String("Italic Text").Italic()
	underline := out.String("Underlined Text").Underline()
	
	fmt.Printf("Bold: %s\n", bold)
	fmt.Printf("Italic: %s\n", italic)
	fmt.Printf("Underline: %s\n", underline)
	
	// Test background colors - should render as plain text
	bgRed := out.String("Background Red").Background(termenv.RGBColor("#FF0000"))
	
	fmt.Printf("Background: %s\n", bgRed)
}

File: test/corpus/component/04-no-color-ascii/case.ts
import {
  ansi256Color,
  ansiColor,
  newOutput,
  Profile,
  rgbColor,
  withProfile,
} from '../../../../src/index.js';

// Force Ascii profile for consistent testing
const out = newOutput(process.stdout, withProfile(Profile.Ascii));

// Test NoColor with various color attempts
console.log('--- NoColor Tests (ASCII Profile) ---');

// Test with RGB colors - should render as plain text
const red = out.string('Red Text').foreground(rgbColor('#FF0000'));
const blue = out.string('Blue Text').foreground(rgbColor('#0000FF'));
const green = out.string('Green Text').foreground(rgbColor('#00FF00'));

console.log(`RGB Red: ${red.toString()}`);
console.log(`RGB Blue: ${blue.toString()}`);
console.log(`RGB Green: ${green.toString()}`);

// Test with ANSI colors - should render as plain text
const ansiRed = out.string('ANSI Red').foreground(ansiColor(1));
const ansiBlue = out.string('ANSI Blue').foreground(ansiColor(4));

console.log(`ANSI Red: ${ansiRed.toString()}`);
console.log(`ANSI Blue: ${ansiBlue.toString()}`);

// Test with ANSI256 colors - should render as plain text
const ansi256Red = out.string('ANSI256 Red').foreground(ansi256Color(196));
const ansi256Blue = out.string('ANSI256 Blue').foreground(ansi256Color(21));

console.log(`ANSI256 Red: ${ansi256Red.toString()}`);
console.log(`ANSI256 Blue: ${ansi256Blue.toString()}`);

// Test styling - should also render as plain text
const bold = out.string('Bold Text').bold();
const italic = out.string('Italic Text').italic();
const underline = out.string('Underlined Text').underline();

console.log(`Bold: ${bold.toString()}`);
console.log(`Italic: ${italic.toString()}`);
console.log(`Underline: ${underline.toString()}`);

// Test background colors - should render as plain text
const bgRed = out.string('Background Red').background(rgbColor('#FF0000'));

console.log(`Background: ${bgRed.toString()}`);
